Indicators/AwesomeOscillator.cs

```diff
@@ -0,0 +1,101 @@
+    {
+        /// <summary>
+        /// Gets the indicators slow period moving average.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> SlowAo { get; }
+
+        /// <summary>
+        /// Gets the indicators fast period moving average.
+        /// </summary>
+        private IndicatorBase<IndicatorDataPoint> FastAo { get; }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => SlowAo.IsReady && FastAo.IsReady;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Creates a new Awesome Oscillator from the specified periods.
+        /// </summary>
+        /// <param name="fastPeriod">The period of the fast moving average associated with the AO</param>
+        /// <param name="slowPeriod">The period of the slow moving average associated with the AO</param>
+        /// <param name="type">The type of moving average used when computing the fast and slow term. Defaults to simple moving average.</param>
+        public AwesomeOscillator(int fastPeriod, int slowPeriod, MovingAverageType type=MovingAverageType.Simple)
+            : this($"AO({fastPeriod},{slowPeriod},{type})", fastPeriod, slowPeriod, type)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new Awesome Oscillator from the specified periods.
+        /// </summary>
+        /// <param name="name">The name of this indicator</param>
+        /// <param name="fastPeriod">The period of the fast moving average associated with the AO</param>
+        /// <param name="slowPeriod">The period of the slow moving average associated with the AO</param>
+        /// <param name="type">The type of moving average used when computing the fast and slow term. Defaults to simple moving average.</param>
+        public AwesomeOscillator(string name, int fastPeriod, int slowPeriod,  MovingAverageType type=MovingAverageType.Simple)
+            : base(name)
+        {
+            SlowAo = type.AsIndicator(slowPeriod);
+            FastAo = type.AsIndicator(fastPeriod);
+            WarmUpPeriod = Math.Max(slowPeriod, fastPeriod);
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            var presentValue = (input.High + input.Low) / 2;
+            SlowAo.Update(input.Time, presentValue);
+            FastAo.Update(input.Time, presentValue);
+
+            return IsReady ? FastAo.Current.Value - SlowAo.Current.Value : 0m;
```

Comment:
Nit: we can use `return IsReady ? FastAo - SlowAo : 0m;` for short, since the indicator are implicitly converted to its current value (decimal).